fullstack-cert/typescript-projects/flashcard-quiz-app/src/script.ts

```diff
@@ -0,0 +1,226 @@
+const cardDisplay = document.querySelector("#current-card") as HTMLElement;
+const cardButtonsContainer = document.getElementById(
+  "cards-list",
+) as HTMLElement;
+
+const frontInput = document.getElementById("front-text") as HTMLTextAreaElement;
+const backInput = document.getElementById("back-text") as HTMLTextAreaElement;
+
+let currentCardIndex = -1;
+let showingFront = true;
+let currentCards: FlashCard[] = [];
+
+interface FlashCard {
+  frontText: string;
+  backText: string;
+}
+
+class InvalidUserInputError extends Error {
+  constructor(message: string) {
+    super(message);
+    this.name = "InvalidUserInputError";
+    Object.setPrototypeOf(this, InvalidUserInputError.prototype);
+  }
+}
+
+function refresh(): void {
+  if (currentCardIndex < 0 || !currentCards[currentCardIndex]) {
+    cardDisplay.textContent =
+      "No card selected. Add a new card to get started!";
+    return;
+  }
+
+  const card = currentCards[currentCardIndex];
+  cardDisplay.querySelector(".card-front").textContent = card.frontText;
+  cardDisplay.querySelector(".card-back").textContent = card.backText;
+  // add correct background to current card
+  Array.from(cardButtonsContainer.children).forEach((child, i) => {
+    const button = child as HTMLElement;
```

Comment:
We could maybe turn this into a type guard? 

```ts
const isButtonElement = (element: unknown): element is HTMLButtonElement => {
  return element instanceof HTMLButtonElement;
}
```

Something like this - have not tested the `instanceof`.

Reply:
:heavy_check_mark: